src/manifold/src/quickhull.cpp

```diff
@@ -85,34 +228,92 @@ void QuickHull::buildMesh(const VertexDataSource& pointCloud, bool CCW,
   // The planar case happens when all the points appear to lie on a two
   // dimensional subspace of R^3.
   planar = false;
-  createConvexHalfEdgeMesh();
+  createConvexHalfedgeMesh();
   if (planar) {
-    const size_t extraPointIndex = planarPointCloudTemp.size() - 1;
-    for (auto& he : mesh.halfEdges) {
-      if (he.endVertex == extraPointIndex) {
-        he.endVertex = 0;
+    const int extraPointIndex = planarPointCloudTemp.size() - 1;
+    for (auto& he : mesh.halfedges) {
+      if (he.endVert == extraPointIndex) {
+        he.endVert = 0;
       }
     }
     originalVertexData = pointCloud;
     planarPointCloudTemp.clear();
   }
-}
 
-ConvexHull QuickHull::getConvexHull(const VertexDataSource& pointCloud,
-                                    bool CCW, bool useOriginalIndices,
-                                    double epsilon) {
-  buildMesh(pointCloud, CCW, useOriginalIndices, epsilon);
-  return ConvexHull(mesh, originalVertexData, CCW, useOriginalIndices);
+  // reorder halfedges
+  Vec<Halfedge> halfedges(mesh.halfedges.size());
+  Vec<int> counts(mesh.halfedges.size(), 0);
+  Vec<int> mapping(mesh.halfedges.size());
+
+  // reorder halfedges
+  size_t j = 0;  // Since all faces are not used now (so we should start from 0
+                 // and just increment, we can later set the face id as index/3
+                 // for each halfedge)
+  for_each(autoPolicy(mesh.halfedges.size()), countAt(0_uz),
+           countAt(mesh.halfedges.size()), [&](size_t i) {
+             if (mesh.halfedges[i].pairedHalfedge < 0) return;
```

Comment:
This part will not work when run in parallel. You can change the autoPolicy to ExecutionPolicy::Seq for now and optimize that later.